container/sagemaker/proxy_client.py

```diff
@@ -90,13 +77,17 @@ def delete_model(self, model_name):
                         model_server_config.model_config_list.CopyFrom(config_list)
                         req = model_management_pb2.ReloadConfigRequest()
                         req.config.CopyFrom(model_server_config)
-                        self.stub.HandleReloadConfigRequest(req)
+                        self.stub.HandleReloadConfigRequest(request=req,
+                                                            timeout=GRPC_REQUEST_TIMEOUT_IN_SECONDS,
+                                                            wait_for_ready=True)
                         self._delete_model_from_config_file(model_server_config)
 
                 # no such model exists
-                raise Exception(404, '{} not loaded yet.'.format(model_name))
+                raise FileNotFoundError
             except grpc.RpcError as e:
-                raise Exception(e.code(), e.details())
+                if e.code() is grpc.StatusCode.DEADLINE_EXCEEDED:
+                    raise MultiModelException(408, e.details())
+                raise MultiModelException(500, e.detals())
```

Comment:
typo in method name `detals` 

this branch has obviously not been run at all. please add some tests.

Reply:
Added a test case with mal-formatted model_data to get 500 response.